File: persistence/src/main/java/org/devgateway/toolkit/persistence/dto/rainfall/RainLevelChart.java
package org.devgateway.toolkit.persistence.dto.rainfall;

/**
 * @author Octavian Ciubotaru
 */
public class RainLevelChart {

    private final RainLevelChartConfig config;
    private final RainLevelChartFilter filter;
    private final RainLevelChartData data;

    public RainLevelChart(RainLevelChartConfig config, RainLevelChartFilter filter,
            RainLevelChartData data) {
        this.config = config;
        this.filter = filter;
        this.data = data;
    }

    public RainLevelChartConfig getConfig() {
        return config;
    }

    public RainLevelChartFilter getFilter() {
        return filter;
    }

    public RainLevelChartData getData() {
        return data;
    }
}

File: ui20/src/modules/entities/rainfall/RainLevelChart.js
import RainLevelConfig from "./RainLevelConfig"
import RainLevelFilter from "./RainLevelFilter"
import RainLevelData from "./RainLevelData"
import RainLevelSetting from "./RainLevelSetting"

export default class RainLevelChart {
  config: RainLevelConfig
  filter: RainLevelFilter
  data: RainLevelData
  setting: RainLevelSetting

  constructor({ config, filter, data }) {
    this.config = new RainLevelConfig(config)
    this.filter = new RainLevelFilter(filter)
    this.data = new RainLevelData(data)
    this.setting = new RainLevelSetting()
  }
}
